### ISSUE TITLE
Google Play Pre-Launch Report cannot control Flutter PageView

### ISSUE THREAD

**Issue by sceee** [flutter/flutter#35341]
First of all, I am facing this issue when uploading my Flutter app in Google Play and the automatic Pre-Launch Report tries to "test" my app.
The problem is that I have a First-Start Tutorial based on a Pager that is shown automatically at application startup. And it seems that the Pre-Launch Report cannot control the Flutter Pager because it just stays stuck at this screen.

## Steps to Reproduce
1. Have a flutter application that shows a PageView at startup
2. Upload Application Release to Google Play so that the Pre-Launch Report gets started
3. Check the Pre-Launch Report once it's finished

## Logs
Not present from Google Play Pre-Launch Report

There are also errors reported like this for all the tested devices:
Object Label:
Element Path android:id​/content​/FlutterView[0]
This object may not have a label that can be used for reading.

Are Flutter applications supported in Google Play Pre-Launch at all at the moment? I found this related issue but it does not have a definite answer: #31670

**Comment by BondarenkoStas** [comment]
@sceee do you still experience the issue?

**Comment by sceee** [comment]
@BondarenkoStas yes, basically I am still facing this issue. All Pre-Launch devices stay on the first screen which is a Slider (PageView) Page (as a lot of apps have). The first slide does not get slided away. 

But it seems Google Play now somehow detects more of the app because it tries to make random clicks now - that was not happening earlier. Also, the videos from the Pre-Launch reports are way longer than previously. They used to be 30s and now they are several minutes. 
But as said, the app stays on the first screen.

**Comment by BondarenkoStas** [comment]
@eseidelGoogle can you elaborate on it?

**Comment by Stewioie** [comment]
An other problem is my flutter app has a sign in page. I cannot add sign in parameters because I cannot set textfields ids and button id

**Comment by eseidelGoogle** [comment]
FYI @csells @xster

**Comment by xster** [comment]
This is a standard Flutter app you created via `flutter create` right? 

I imagine this is espresso related. If so, this is being currently worked on @adazh, @collinjackson

**Comment by sceee** [comment]
Yes @xster . Of course some time ago but initially, yes.

**Comment by adazh** [comment]
This is an issue we are aware of: the pre-launch tests cannot "see" Flutter views/pages.
However, one of top blockers for the support is that currently we do not have a good way to test a release Flutter app. @collinjackson @tvolkert may have more insight.

**Comment by neiljaywarner** [comment]
@adazh @collinjackson @tvolkert it looks like from seeing https://support.google.com/googleplay/android-developer/answer/7002270?hl=en that maybe we can do debug builds and internal builds and get prelaunch reports?

**Comment by neiljaywarner** [comment]
well, for those following this - perhaps a simple option is to have some if firebase-test-lab/prelaunch report, then autologin-as-test-user.  Yes, it requires a "hack" or "patch" just for this purpose, but should be pretty easy, by checking IP address and/or system property.
 
of course that has nothing to do with a flutter "feature" unless flutter team adds a "isFirebaseTestLab()" method or plugin somewhere :)   also see https://pub.dev/packages/flutter_runtime_env

**Comment by tvolkert** [comment]
@dnfield FYI

**Comment by dnfield** [comment]
The prelaunch documentation says it's powered by Firebase Test Lab. This almost certainly defaults to something like RoboTesting for apps, which won't work for Flutter since it's more like an Open GL game.

You do have the option to run a game loop test though, which you can read more about here: https://firebase.google.com/docs/test-lab/game-loop.  That type of test would work with Flutter without further changes to Flutter, but it's probably not an ideal fit.

Longer term, we may want to see if we can work with the Firebase team to do something like robotesting with Flutter apps.

**Comment by dnfield** [comment]
The more relevant link is https://firebase.google.com/docs/test-lab/android/game-loop - not really sure why the default is for iOS and why the Play Store doc is linking there :)

**Comment by dnfield** [comment]
And while I'm here, a couple very rough thoughts on how this could be automated/made better:

- Turn on a11y services for the app/phone.
- Use the semantics tree information to drive a robo-like test, instead of trying to use the Java view heirarchy information the way a regular robotest works

This has the advantages of not requiring an observatory service for release mode (which would incur several penalities we'd rather not), testing accessibility in the app, and being something that should be as automatable as the current tests.

It's unfortunately not entirely trivial to implement :)

**Comment by neiljaywarner** [comment]
I hear you .I often refer to flutter as unity for apps as well... But..

If the espresso package lets you make bindings and have espresso tests in
the Androidtest directory in kotlin code couldn't that be run from firebase
test lab finding specific flutter widgets ?



....sent from my phone

**Comment by dnfield** [comment]
I don't believe the pre-launch tests support using Espresso. There is a Flutter Espresso package at this point, and it's certainly useful, but it's different from this topic.

At some point, I will try to document a plan for a Flutter-style robo test, which should be possible and would be useful for this kind of thing (e.g. find all the inputs in my view and pretend to be a user pressing them, make sure nothing crashes).

**Comment by neiljaywarner** [comment]
This should work and be almost the same thing with espresso plus flutter
plus test lab plus test matrix right?

https://support.google.com/firebase/answer/6386654?hl=en




....sent from my phone

**Comment by dnfield** [comment]
The instrumentation tests with espresso should work, although I believe that is not the same as the pre-launch tests that Google Play Store runs - if I'm correct, those are more like robotests, which don't work with Flutter (or Unity apps, or games, etc.).
